packages/tupaia-web/src/features/Dashboard/Dashboard.tsx

```diff
@@ -103,17 +104,33 @@ const DashboardImageContainer = styled.div`
 `;
 
 export const Dashboard = () => {
+  const navigate = useNavigate();
   const { projectCode, entityCode, dashboardName } = useParams();
-  const { dashboards, activeDashboard } = useDashboards(projectCode, entityCode, dashboardName);
+  const { data: project, isLoading: isLoadingProject } = useProject(projectCode);
+  const { dashboards, activeDashboard, isLoading: isLoadingDashboards } = useDashboards(
+    projectCode,
+    entityCode,
+    dashboardName,
+  );
   const [isExpanded, setIsExpanded] = useState(false);
   const [exportModalOpen, setExportModalOpen] = useState(false);
   const { data: entity } = useEntity(projectCode, entityCode);
   const bounds = entity?.bounds || DEFAULT_BOUNDS;
+  const defaultEntityLink = useEntityLink(entityCode);
 
   const toggleExpanded = () => {
     setIsExpanded(!isExpanded);
   };
 
+  // check for valid dashboard name, and if not valid and not still loading, redirect to default dashboard
+  const dashboardNotFound =
```

Comment:
I think this is the best place for it but I was getting an error about max renders exceeded which was caused by one of the useEffect dependencies so I refactored it slightly. As far as I can tell it has the same outcome